subgraphs/_reference_/src/common/metrics.ts

```diff
@@ -0,0 +1,71 @@
+import { BigDecimal, Address, ethereum } from "@graphprotocol/graph-ts";
+import { _Account, _DailyActiveAccount, UsageMetricsDailySnapshot } from "../../generated/schema";
+import { FACTORY_ADDRESS, SECONDS_PER_DAY } from "./constants";
+import { getOrCreateDexAmm, getOrCreatePoolDailySnapshot, getOrCreateUsageMetricSnapshot } from "./getters";
+
+// These are meant more as boilerplates that'll be filled out depending on the
+// subgraph, and will be different from subgraph to subgraph, hence left
+// partially implemented and commented out.
+// They are common within a subgraph but not common across different subgraphs.
+
+// Update FinancialsDailySnapshots entity
+export function updateFinancials(event: ethereum.Event): void {
+  // let financialMetrics = getOrCreateFinancials(event);
+  // let protocol = getOrCreateDexAmm();
+  // // Update the block number and timestamp to that of the last transaction of that day
+  // financialMetrics.blockNumber = event.block.number;
+  // financialMetrics.timestamp = event.block.timestamp;
+  // financialMetrics.totalValueLockedUSD = protocol.totalValueLockedUSD;
+  // ...
+  // financialMetrics.save();
+}
+
+export function updateUsageMetrics(event: ethereum.Event, from: Address): void {
```

Comment:
Can we make it more generalized, so that it can handle both events for event Handlers and Calls for call Handlers